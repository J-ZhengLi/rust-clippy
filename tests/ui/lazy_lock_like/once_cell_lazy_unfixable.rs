//@aux-build:once_cell.rs
//@no-rustfix

#![warn(clippy::once_cell_lazy)]
#![allow(static_mut_refs)]

use once_cell::sync::Lazy;

fn main() {}

static LAZY_FOO: Lazy<String> = Lazy::new(|| "foo".to_uppercase());
//~^ ERROR: this type has been superceded by `std::sync::LazyLock`
static LAZY_BAR: Lazy<String> = Lazy::new(|| "bar".to_uppercase());
//~^ ERROR: this type has been superceded by `std::sync::LazyLock`
static mut LAZY_BAZ: Lazy<String> = Lazy::new(|| "baz".to_uppercase());
//~^ ERROR: this type has been superceded by `std::sync::LazyLock`

fn calling_irreplaceable_fns() {
    let _ = Lazy::get(&LAZY_BAR);

    unsafe {
        let _ = Lazy::get_mut(&mut LAZY_BAZ);
        let _ = Lazy::force_mut(&mut LAZY_BAZ);
    }
}
