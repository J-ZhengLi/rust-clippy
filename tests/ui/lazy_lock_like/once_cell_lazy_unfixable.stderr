error: this type has been superceded by `std::sync::LazyLock`
  --> tests/ui/lazy_lock_like/once_cell_lazy_unfixable.rs:11:18
   |
LL | static LAZY_FOO: Lazy<String> = Lazy::new(|| "foo".to_uppercase());
   |                  ^^^^
   |
   = note: `-D clippy::once-cell-lazy` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::once_cell_lazy)]`
help: use `std::sync::LazyLock` instead
   |
LL | static LAZY_FOO: std::sync::LazyLock<String> = std::sync::LazyLock::new(|| "foo".to_uppercase());
   |                  ~~~~~~~~~~~~~~~~~~~           ~~~~~~~~~~~~~~~~~~~~~~~~

error: this type has been superceded by `std::sync::LazyLock`
  --> tests/ui/lazy_lock_like/once_cell_lazy_unfixable.rs:13:18
   |
LL | static LAZY_BAR: Lazy<String> = Lazy::new(|| "bar".to_uppercase());
   |                  ^^^^
   |
help: use `std::sync::LazyLock` instead
   |
LL | static LAZY_BAR: std::sync::LazyLock<String> = std::sync::LazyLock::new(|| "bar".to_uppercase());
   |                  ~~~~~~~~~~~~~~~~~~~           ~~~~~~~~~~~~~~~~~~~~~~~~

error: this type has been superceded by `std::sync::LazyLock`
  --> tests/ui/lazy_lock_like/once_cell_lazy_unfixable.rs:15:22
   |
LL | static mut LAZY_BAZ: Lazy<String> = Lazy::new(|| "baz".to_uppercase());
   |                      ^^^^
   |
help: use `std::sync::LazyLock` instead
   |
LL | static mut LAZY_BAZ: std::sync::LazyLock<String> = std::sync::LazyLock::new(|| "baz".to_uppercase());
   |                      ~~~~~~~~~~~~~~~~~~~           ~~~~~~~~~~~~~~~~~~~~~~~~

error: aborting due to 3 previous errors

